framework/src/outputs/BlockRestrictionDebugOutput.C

```diff
@@ -0,0 +1,310 @@
+#include "MooseApp.h"
+#include "Material.h"
+#include "ConsoleUtils.h"
+#include "MooseMesh.h"
+#include "MooseObjectName.h"
+#include "NonlinearSystemBase.h"
+#include "MooseVariableBase.h"
+#include "BlockRestrictable.h"
+#include "KernelBase.h"
+#include "AuxiliarySystem.h"
+#include "AddKernelAction.h"
+#include "AuxKernel.h"
+#include "ConsoleUtils.h"
+
+#include "libmesh/transient_system.h"
+
+registerMooseObject("MooseApp", BlockRestrictionDebugOutput);
+
+InputParameters
+BlockRestrictionDebugOutput::validParams()
+{
+  InputParameters params = Output::validParams();
+
+  params.addParam<MultiMooseEnum>("scope",
+                                  getScopes("all"),
+                                  "The types of object to output block coverage for, if nothing is "
+                                  "provided everything will be output.");
+
+  params.addParam<NonlinearSystemName>(
+      "nl_sys", "nl0", "The nonlinear system that we should output information for.");
+
+  params.addClassDescription(
+      "Debug output object for displaying information regarding block-restriction of objects.");
+  params.set<ExecFlagEnum>("execute_on") = EXEC_INITIAL;
+  return params;
+}
+
+// Defines the scopes to enable for the BlockRestrictionDebugOutput object
+MultiMooseEnum
+BlockRestrictionDebugOutput::getScopes(std::string default_scopes)
+{
+  return MultiMooseEnum("none all variables kernels auxvariables auxkernels materials userobjects",
+                        default_scopes);
+}
+
+BlockRestrictionDebugOutput::BlockRestrictionDebugOutput(const InputParameters & parameters)
+  : Output(parameters),
+    _scope(getParam<MultiMooseEnum>("scope")),
+    _nl(_problem_ptr->getNonlinearSystemBase(
+        _problem_ptr->nlSysNum(getParam<NonlinearSystemName>("nl_sys")))),
+    _sys(_nl.system())
+{
+}
+
+void
+BlockRestrictionDebugOutput::output()
+{
+  printBlockRestrictionMap();
+}
+
+void
+BlockRestrictionDebugOutput::printBlockRestrictionMap() const
+{
+  // is there anything to do?
+  if (_scope.isValid() && _scope.contains("none"))
+    return;
+
+  // Build output stream
+  std::stringstream out;
+
+  auto printCategoryAndNames = [&out](std::string category, std::set<std::string> & names)
+  {
+    const auto n = names.size();
+    if (n > 0)
+    {
+      // print the category and number of items
+      out << "    " << category << " (" << n << " " << ((n == 1) ? "item" : "items") << "): ";
+
+      // If we would just use Moose::stringify(names), the indention is not right.
+      // So we are printing the names one by one and using ConsoleUtils::insertNewline.
+      std::streampos begin_string_pos = out.tellp();
+      std::streampos curr_string_pos = begin_string_pos;
+      unsigned int i = 0;
+      for (const auto & name : names)
+      {
+        out << Moose::stringify(name) << ((i++ < (n - 1)) ? ", " : "");
+        curr_string_pos = out.tellp();
+        ConsoleUtils::insertNewline(out, begin_string_pos, curr_string_pos);
```

Comment:
so you didnt like the output of this?
```
  std::string list_kernels = ConsoleUtils::mooseObjectVectorToString(objs_ptrs);
  console << ConsoleUtils::formatString(list_kernels, "[DBG]") << std::endl;
```

Reply:
This is what I tried following your comment. But with `ConsoleUtils::formatString`, the overflow was not indented for me. Since I assume that indentation improves readability, I have reverted to ConsoleUtils::insertNewline. 

Ultimately, I am indifferent as to which function is used for the output. Please you choose.

Reply:
works for me.
Maybe I should add the overflow indent to formatString when I have time